src/main/java/Duke.java

```diff
@@ -1,10 +1,342 @@
+     */
+    private static void executeListAllTasks() {
+        displayToUser(tasks);
+    }
+
+    /**
+     * Checks if task number is valid. 
+     * 
+     * @param taskNumber Task number. 
+     * @return True if number is valid and false if invalid. 
+     */
+    private static boolean isTaskNumberValid(int taskNumber) {
+        return taskNumber >= 0 && taskNumber < sizeOfTaskList;
+    }
+
+    /**
+     * Marks a given task number as done. 
+     * 
+     * @param taskNumber The task number to mark as done (starting with 1). 
+     */
+    private static void executeMarkTask(int taskNumber) {
+        tasks[taskNumber].setTaskStatus();
+    }
+
+    /**
+     * Displays the success message after marking task as done. 
+     * 
+     * @param taskNumber The task number (starting from 1) that was marked as done. 
+     */
+    private static void displayMarkTaskSuccessMessage(int taskNumber) {
+        displayToUser(MESSAGE_MARKED, tasks[taskNumber].toString());
+    }
+
+    /**
+     * Adds a new todo task to task list. 
+     */
+    private static void executeAddTodo() {
+        tasks[sizeOfTaskList] = new Todo(userCommand);
+        sizeOfTaskList++;
+        displayAddTaskSuccessMessage();
+    }
+
+    /**
+     * Adds a new deadline task to task list. 
+     * Processes the user input to extract date. 
+     * If date is not null, create new deadline task. 
+     * Otherwise, display invalid command error. 
+     */
+    private static void executeAddDeadline() {
+        String date = processParameters(DEADLINE_DATA_PREFIX_BY);
+        if (date != null) {
+            tasks[sizeOfTaskList] = new Deadline(userCommand, date);
+            sizeOfTaskList++;
+            displayAddTaskSuccessMessage();
+        } else {
+            displayToUser(ERROR_INVALID_COMMAND_RECEIVED);
+        }
+    }
+
+    /**
+     * Adds a new event task to task list. 
+     */
+    private static void executeAddEvent() {
+        // Process event parameters
+        String date = processParameters(EVENT_DATA_PREFIX_AT);
+        if (date != null) {
+            tasks[sizeOfTaskList] = new Event(userCommand, date);
+            sizeOfTaskList++;
+            displayAddTaskSuccessMessage();
+        } else {
+            // Display error message
+            displayToUser(ERROR_INVALID_COMMAND_RECEIVED);
+        }
+
+    }
+
+    /**
+     * Finds the location of the filters and locate the date from the task. 
+     * Removes the task date information from userCommand, keeping only the task name in userCommand. 
+     * 
+     * @param filterString The string to find in userCommand depending on an event (/at) or deadline (/by). 
+     * @return The date extracted from userCommand. 
+     */
+    private static String processParameters(String filterString) {
+        int indexOfDate = userCommand.indexOf(filterString);
+        if (indexOfDate > 0) {
+            String date = userCommand.substring(indexOfDate + 4);
```

Comment:
You may want to use a constant to replace the number 4 here (or maybe a comment to explain) as other's may not understand why we have to offset 4 from the date index itself.

Reply:
That's true. I will leave a comment indicating the magic number. Thanks.